src/client/AstarApi.ts

```diff
@@ -36,6 +42,39 @@ export class AstarApi implements IAstarApi {
         return decimals[0];
     }
 
+    public async getAprCalculationData(): Promise<AprCalculationData> {
+        const oneBillion = 1000000000;
+
+        await this.connect();
+        const results = await Promise.all([
+            this._api.consts.blockReward.rewardAmount,
+            this._api.query.timestamp.now(),
+            this._api.rpc.chain.getHeader(),
+            this._api.consts.dappsStaking.developerRewardPercentage,
+            this._api.consts.dappsStaking.blockPerEra,
+        ]);
+
+        const result = new AprCalculationData();
+        result.blockRewards = results[0] as u128;
+        result.timeStamp = results[1];
+        result.latestBlock = (results[2] as Header).number.unwrap();
+
+        // The value below is of Perbill type, so we need to divide by billion to get percentage.
+        result.developerRewardPercentage = (results[3] as UInt).toNumber() / oneBillion;
```

Comment:
correct me if I'm wrong, but this division doesn't look too safe in my opinion (especially with the `.toNumber()` casting).
Is there a safer way of handling these types of calculations?

Reply:
good point, for some stupid reason I thought that result[3] can't be higher than one billion, but why shouldn't be. Fixed